application/forms/EventRuleConfigForm.php

```diff
@@ -0,0 +1,647 @@
+
+                continue;
+            }
+
+            $escalationCondition = new EscalationCondition($prefixMap, $this);
+            $escalationRecipient = new EscalationRecipient($prefixMap);
+            $this->registerElement($escalationCondition);
+            $this->registerElement($escalationRecipient);
+
+            $escalation = new Escalation(
+                $escalationCondition,
+                $escalationRecipient,
+                $removeEscalationButtons[$prefixMap] ?? null
+            );
+
+            if ($zeroConditionEscalation === $prefixMap && $escalation->addConditionHasBeenPressed()) {
+                $this->hasZeroConditionEscalation = false;
+                $zeroConditionEscalation = null;
+            } elseif ($escalation->lastConditionHasBeenRemoved()) {
+                $this->hasZeroConditionEscalation = true;
+                $zeroConditionEscalation = $prefixMap;
+            }
+
+            $escalations[] = $escalation;
+        }
+
+        $this->getElement('zero-condition-escalation')->setValue($zeroConditionEscalation);
+
+        $this->addHtml(
+            (new HtmlElement('div', Attributes::create(['class' => 'filter-wrapper'])))
+                ->addHtml(
+                    (new FlowLine())->getRightArrow(),
+                    $configFilter,
+                    (new FlowLine())->getHorizontalLine()
+                )
+        );
+
+        $escalationWrapper = (new HtmlElement('div'))->addHtml(new Escalations($escalations), $addEscalationButton);
+        $this->addHtml($escalationWrapper);
+    }
+
+    /**
+     * Handle addition of escalations
+     *
+     * @return void
+     */
+    protected function handleAdd(): void
+    {
+        $pressedButton = $this->getPressedSubmitElement();
+
+        if ($pressedButton && $pressedButton->getName() === 'add-escalation') {
+            $this->clearPopulatedValue('prefixes-map');
+            $prefixesMapString = $this->getValue('prefixes-map', '');
+            $prefixesMap = explode(',', $prefixesMapString);
+            $escalationFakePos = bin2hex(random_bytes(4));
+            $prefixesMap[] = $escalationFakePos;
+            $this->getElement('prefixes-map')
+                ->setValue(implode(',', $prefixesMap));
+
+            if ($this->getValue('zero-condition-escalation') === null) {
+                $this->getElement('zero-condition-escalation')
+                    ->setValue($escalationFakePos);
+            }
+        }
+    }
+
+    public function populate($values): self
+    {
+        if (! isset($values['rule_escalation'])) {
+            return parent::populate($values);
+        }
+
+        $formValues = [];
+        $formValues['prefixes-map'] = $this->getPrefixesMap(count($values['rule_escalation']));
+
+        foreach ($values['rule_escalation'] as $position => $escalation) {
+            $conditions = explode('|', $escalation['condition'] ?? '');
```

Comment:
It's odd to me that these are parsed here. For one, in `getValues()` they aren't built, so that must be done somewhere else, so why are they parsed here? And second, they're parsed *by hand*, meaning, not by `QueryString::parse`. I hope this is just a mistake here, and where ever they're built, this is not also done *by hand*.